sdk/core/azure-core/azure/core/_pipeline_client_async.py

```diff
@@ -189,6 +209,7 @@ async def _make_pipeline_call(self, request, **kwargs):
                     # the body is loaded. instead of doing response.read(), going to set the body
                     # to the internal content
                     rest_response._content = response.body()  # pylint: disable=protected-access
+                    await rest_response._set_read_checks()  # pylint: disable=protected-access
```

Comment:
Seems like _set_read_checks() closes the response. Please double check.

Reply:
sounds good, since close is in set_read_checks, i'll remove the extra close I have here